openedx/tests/completion_integration/test_models.py

```diff
@@ -56,7 +56,7 @@ def setUp(self):
         self.set_up_completion()
 
     def test_changed_value(self):
-        with self.assertNumQueries(4):  # Get, update, 2 * savepoints
+        with self.assertNumQueries(6):  # Get, update, 2 * savepoints, user exists, completion exists
```

Comment:
I'm not too comfortable with having this magic number as-is, I think it's better to be written as:

```
SELECT = 1  # Represents a select query
UPDATE = 1 # Represents an update query
SAVEPOINT = 1 # Represent all start transaction, a commit or a rollback
DUMMY = 1  # The test framework inserts dummy queries when .... (please add details) .... so we're counting them

...

1*SELECT + 1*UPDATE + 2*SAVEPOINT +  2*DUMMY
```

Given those instances are defined somewhere in the class with descriptive comments.

Reply:
I see that everywhere we're adding `2*SAVEPOINT`, except for this one, in which we add `2*DUMMY`

I'm a bit confused could you please explain?

Reply:
That does seem odd.  I'm not sure exactly why.  It's possible that the previous numbers were incorrectly reported.  The two extra queries are not "dummy queries," but the kind of queries that get made when someone calls `.exists()`.  (`SELECT 1 FROM table <WHERE condition> LIMIT 1`).